autoload/pandoc/toc.vim

```diff
@@ -59,33 +60,33 @@ function! pandoc#toc#Update()
 endfunction
 
 " ReDisplay(bufname): Prepare the location list window for our uses {{{1
-function! pandoc#toc#ReDisplay(bufname)
+function! pandoc#toc#ReDisplay(bufname) abort
     if len(getloclist(0)) == 0
         lclose
         return
     endif
     let &winwidth=(&columns/3)
-    execute "setlocal statusline=pandoc#TOC:".escape(a:bufname, ' ')
+    execute 'setlocal statusline=pandoc#TOC:'.escape(a:bufname, ' ')
 
     " change the contents of the location-list buffer
     set modifiable
-    silent %s/\v^([^|]*\|){2,2} #//e
-    for l in range(1, line("$"))
+    call substitute('%', '\v^([^|]*\|){2,2} #', '')
```

Comment:
Of all the changes here, this is the one I am least confident of. Using `:s///` is subject to all kinds of user mode settings, and the `substitute()` command eliminates those but it also handles magic modes differently (`\v`) and I wasn't sure how to actually test that this is still doing what it was supposed to.